src/devices/Pn5180/Pn5180.cs

```diff
@@ -0,0 +1,1496 @@
+
+            var ret = ReadEeprom(EepromAddress.ProductVersion, versionAnswer);
+            if (!ret)
+            {
+                return (null, null, null);
+            }
+
+            var product = new Version(versionAnswer[1], versionAnswer[0]);
+            var firmware = new Version(versionAnswer[3], versionAnswer[2]);
+            var eeprom = new Version(versionAnswer[5], versionAnswer[4]);
+            return (product, firmware, eeprom);
+        }
+
+        /// <summary>
+        /// Get the PN5180 identifier, this is a 16 byte long
+        /// </summary>
+        /// <param name="identifier">A 16 byte buffer</param>
+        /// <returns>True if success</returns>
+        public bool GetIdentifier(Span<byte> identifier)
+        {
+            if (identifier.Length != 16)
+            {
+                throw new ArgumentException($"Identifier must be 16 bytes long");
+            }
+
+            return ReadEeprom(EepromAddress.DieIdentifier, identifier);
+        }
+
+        /// <summary>
+        /// Read the full EEPROM
+        /// </summary>
+        /// <param name="eeprom">At least 255 bytes buffer</param>
+        /// <returns>True if success</returns>
+        public bool ReadAllEeprom(Span<byte> eeprom)
+        {
+            if (eeprom.Length > 255)
+            {
+                throw new ArgumentException($"Size of EEPROM is 255 bytes, {nameof(eeprom)} must be less than 255 bytes");
+            }
+
+            return ReadEeprom(EepromAddress.DieIdentifier, eeprom);
+        }
+
+        /// <summary>
+        /// Write all the EEPROM
+        /// </summary>
+        /// <param name="eeprom">A 255 bytes buffer</param>
+        /// <returns>True if success</returns>
+        public bool WriteAllEeprom(Span<byte> eeprom)
+        {
+            if (eeprom.Length != 255)
+            {
+                throw new ArgumentException($"Size of EEPROM is 255 bytes, {nameof(eeprom)} must be 255 bytes");
+            }
+
+            return WriteEeprom(EepromAddress.DieIdentifier, eeprom);
+        }
+
+        /// <summary>
+        /// Read a specific part of the EEPROM
+        /// </summary>
+        /// <param name="address">The EEPROM address</param>
+        /// <param name="eeprom">A span of byte to read the EEPROM</param>
+        /// <returns>True if success</returns>
+        public bool ReadEeprom(EepromAddress address, Span<byte> eeprom)
+        {
+            if ((byte)address + eeprom.Length > 255)
+            {
+                throw new ArgumentException($"Size of EEPROM is 255 bytes, {nameof(address)} + {nameof(eeprom)} must be less than 255");
+            }
+
+            Span<byte> dumpEeprom = stackalloc byte[3];
+            dumpEeprom[0] = (byte)Command.READ_EEPROM;
+            dumpEeprom[1] = (byte)address;
+            dumpEeprom[2] = (byte)eeprom.Length;
+            LogInfo.Log($"{nameof(ReadEeprom)}, {nameof(dumpEeprom)}: {BitConverter.ToString(dumpEeprom.ToArray())}", LogLevel.Debug);
```

Comment:
slightly bothersome that people opting out of logging pay the ToArray and ToString cost as well. Fine in this PR. If we ever get complain about this or memory related with usage of this library then we can fix.

Reply:
I'll check the log level and call the Log function if log level is passed. a comparison will be much less than a conversion. Will fix everywhere

Reply:
So something like this will make it:
```
if (LogLevel >= LogLevel.Debug)
            {
                LogInfo.Log($"{nameof(ReadEeprom)}, {nameof(dumpEeprom)}: {BitConverter.ToString(dumpEeprom.ToArray())}", LogLevel.Debug);
            }
```